lt/client/client.go

```diff
@@ -423,57 +434,72 @@ func (u *User) transmitSpeech() {
 		}
 
 		for text := range u.speechTextCh {
-			log.Printf("%s: received text to speak: %q", u.cfg.Username, text)
-
-			rd, rate, err := textToSpeech(text)
-			if err != nil {
-				log.Printf("%s: textToSpeech failed: %s", u.cfg.Username, err.Error())
-				continue
-			}
+			func() {
+				defer func() {
+					u.doneSpeakingCh <- struct{}{}
+				}()
+				log.Printf("%s: received text to speak: %q", u.cfg.Username, text)
+
+				var rd io.Reader
+				var rate int
+				var err error
+				if u.pollySession != nil {
+					rd, rate, err = u.pollyToSpeech(text)
+				} else {
+					rd, rate, err = textToSpeech(text)
+				}
+				if err != nil {
+					log.Printf("%s: textToSpeech failed: %s", u.cfg.Username, err.Error())
+					return
+				}
 
-			log.Printf("%s: raw speech samples decoded (%d)", u.cfg.Username, rate)
+				log.Printf("%s: raw speech samples decoded (%d)", u.cfg.Username, rate)
 
-			audioSamplesDataBuf := bytes.NewBuffer([]byte{})
-			if _, err := audioSamplesDataBuf.ReadFrom(rd); err != nil {
-				log.Printf("%s: failed to read samples data: %s", u.cfg.Username, err.Error())
-				continue
-			}
+				audioSamplesDataBuf := bytes.NewBuffer([]byte{})
+				if _, err := audioSamplesDataBuf.ReadFrom(rd); err != nil {
+					log.Printf("%s: failed to read samples data: %s", u.cfg.Username, err.Error())
+					return
```

Comment:
Was this change meant (returning vs continuing)?

Reply:
Yep, because it's in a func now.